tests/test_search.py

```diff
@@ -341,3 +344,20 @@ def assert_is_sorted(item, path):
     # Pretty print the index. Only shown by pytest on failure.
     print(f'searchindex.js contents:\n\n{json.dumps(index, indent=2)}')
     assert_is_sorted(index, '')
+
+
+@pytest.mark.parametrize('directory', (
+    directory for directory in (TESTS_ROOT / 'js' / 'roots').iterdir()
+))
+def test_check_js_search_indexes(make_app, sphinx_test_tempdir, directory):
+    app = make_app('html', srcdir=directory, builddir=sphinx_test_tempdir / directory.name)
+    app.build()
+
+    fresh_searchindex = (app.outdir / 'searchindex.js')
+    fresh_hash = sha256(fresh_searchindex.read_bytes()).digest()
+
+    existing_searchindex = (TESTS_ROOT / 'js' / 'fixtures' / directory.name / 'searchindex.js')
+    existing_hash = sha256(existing_searchindex.read_bytes()).digest()
```

Comment:
Some caution about these lines: I _thought_ that using `pathlib.Path` files' `read_bytes()` and hashing the results of that should produce identical results on different platforms.

A [test failure](https://github.com/sphinx-doc/sphinx/actions/runs/8563318800/job/23468161852#step:6:2342) on Windows where differences appear with very similar code (from an otherwise-unrelated PR) is making me want to double-check that.

Reply:
Ok, in the case of `searchindex.js` this appears not to be a problem - I believe due to the absence of newlines in the file content (meaning that the byte-for-byte content of the file is unchanged when written on different operating system platforms).